tests/test_view_helpers.py

```diff
@@ -0,0 +1,137 @@
+from pytest import fixture
+
+from sqlalchemy import Column, Integer, String, ForeignKey
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.sql import select, text, literal_column
+from sqlalchemy.sql import column
+
+from pycds.util import generic_sesh
+from pycds.view_helpers import ViewMixin
+
+# Define some simple ORM objects (and their tables) to test view helpers against
+
+Base = declarative_base()
+metadata = Base.metadata
+
+
+class Thing(Base):
+    __tablename__ = 'things'
+    id = Column(Integer, primary_key=True)
+    name = Column(String)
+    description_id = Column(Integer, ForeignKey('descriptions.id'))
+
+    def __repr__(self):
+        return '<Thing(id=%d, name=%s, desc_id=%d)>' % (self.id, self.name, self.description_id)
+
+
+class Description(Base):
+    __tablename__ = 'descriptions'
+    id = Column(Integer, primary_key=True)
+    desc = Column(String)
+
+    def __repr__(self):
+        return '<Description(id=%d, desc=%s)>' % (self.id, self.desc)
+
+
+class SimpleThing(Base, ViewMixin):
+    # __selectable__ = select([Thing]).where(Thing.id <= literal_column('3'))  # deeelightful
+
+    # less delightful but applicable to cases where we use text queries for selectable:
+    __selectable__ = text("""
+        SELECT *
+        FROM things
+        WHERE things.id <= 3
+    """).columns(Thing.id, Thing.name, Thing.description_id)
+
+    def __repr__(self):
+        return '<SimpleThing(id=%d, desc=%s)>' % (self.id, self.name)
+
+
+class ThingWithDescription(Base, ViewMixin):
+    # this works:
+    # __selectable__ = select([Thing.id, Thing.name, Description.desc])\
+    #     .select_from(Thing.__table__.join(Description.__table__, Thing.description_id == Description.id))
+
+    # this works:
+    # __selectable__ = text('''
+    #     SELECT things.id, things.name, descriptions.desc
+    #     FROM things JOIN descriptions ON (things.description_id = descriptions.id)
+    # ''').columns(Thing.id, Thing.name, Description.desc)
+
+    # so does this! which is essential for text-defined queries with new columns
+    __selectable__ = text('''
+        SELECT things.id, things.name, descriptions.desc
+        FROM things JOIN descriptions ON (things.description_id = descriptions.id)
+    ''').columns(column('id'), column('name'), column('desc'))
+    __primary_key__ = ['id']
+
+
+class ThingCount(Base, ViewMixin):
+    __selectable__ = text("""
+        SELECT d.desc as desc, count(things.id) as num
+        FROM things JOIN descriptions as d ON (things.description_id = d.id)
+        GROUP BY d.desc
+    """).columns(column('desc'), column('num'))
+    __primary_key__ = ['desc']
+
+    def __repr__(self):
+        return '<ThingWithDescription(id=%d, name=%s, desc=%s)>' % (self.id, self.name, self.desc)
```

Comment:
Yes, since then I've changed over to `str.format()` I believe `%` is deprecated.